### PR TITLE
Fix Geo vis IP address & SQL logic bugs.

### PR DESCRIPTION
**Pull request by lihuoran**
# Description

**Fix Geo vis IP address & SQL logic bugs.**

As reported in [352](https://github.com/microsoft/maro/issues/352), the Geo vis frontend app could not access the database if the database is deployed on a remote server. The root cause is that both the frontend and backend app run in local mode (can only be accessed from the same machine), therefore all external access will be blocked. This PR fixes these bugs.

This PR also fixes some related SQL logic bugs. The original SQL logic fails to get the data under some corner cases.

## Linked issue(s)/Pull request(s)

- [352](https://github.com/microsoft/maro/issues/352)

## Type of Change

- [x] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit
- [x] Geo vis

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [x] Linux
- Python version:
  - [ ] 3.6
  - [x] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [x] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/cli/maro_real_time_vis/back_end/vis_app/data_process/request/utils.py [hunk 1]
```diff
@@ -34,12 +34,4 @@ def get_input_range(start_tick: str, end_tick: str) -> str:
...
-    i = int(start_tick)
-    input_range = "("
-    while i < int(end_tick):
-        if i == int(end_tick) - 1:
-            input_range = input_range + "'" + str(i) + "'" + ")"
-        else:
-            input_range = input_range + "'" + str(i) + "'" + ","
-        i = i + 1
-    return input_range
+    return "(" + ", ".join([f"'{i}'" for i in range(int(start_tick), int(end_tick))]) + ")"
```
Comment (Jinyu-W) [inline maro/cli/maro_real_time_vis/back_end/vis_app/data_process/request/utils.py:37]: newly added space ' '? for better view only or fix some issues?

File: maro/cli/maro_real_time_vis/start_maro_geo_vis.py [hunk 2]
```diff
@@ -77,15 +81,12 @@ def start_geo_vis(start: str, experiment_name: str, front_end_port: int, **kwarg
...
         exec_path = os.path.dirname(inspect.getfile(inspect.currentframe()))
-        os.system("docker pull maro2020/geo_front_service")
+        os.system(f"docker pull {GEO_FRONT_SERVICE_DOCKER_IMAGE}")
         os.system("docker stop geo-vis")
         os.system("docker rm geo-vis")
-        if front_end_port is not None:
-            os.system(f"docker run -d -p {front_end_port}:8080 --name geo-vis maro2020/geo_front_service")
-        else:
-            os.system("docker run -d -p 8080:8080 --name geo-vis maro2020/geo_front_service")
+        os.system(f"docker run -d -p {front_end_port if front_end_port is not None else DEFAULT_FRONT_END_PORT}:8080 "
```
Comment (Jinyu-W) [inline maro/cli/maro_real_time_vis/start_maro_geo_vis.py:87]: Is it the key change for issue #352?

**Review by Jinyu-W — CHANGES_REQUESTED** [review]
- There are still some lint issues
- More description of this PR would be better
- Will you include the fix for issue #314 in this PR later or are you going to open another PR for it?

File: maro/cli/maro_real_time_vis/back_end/vis_app/data_process/request/utils.py (hunk 1 above)
Comment (lihuoran) [reply maro/cli/maro_real_time_vis/back_end/vis_app/data_process/request/utils.py:37]: For better code style only.

File: maro/cli/maro_real_time_vis/start_maro_geo_vis.py (hunk 2 above)
Comment (lihuoran) [reply maro/cli/maro_real_time_vis/start_maro_geo_vis.py:87]: No. The key change for issue [352](https://github.com/microsoft/maro/issues/352) is the IP related configurations. I modified two settings, in both frontend and backend, to enable the external access. The first one is in `maro/cli/maro_real_time_vis/back_end/vis_app/app.py` and the second one is in `maro_vis` (the backend docker).

**Comment by lihuoran** [comment]
> * There are still some lint issues
> * More description of this PR would be better
> * Will you include the fix for issue [🐛 I had an error executing "maro inspector geo --start service..." #314](https://github.com/microsoft/maro/issues/314) in this PR later or are you going to open another PR for it?

- Fixed~
- Enriched~
- @Meroy9819 has made some efforts on this issue before and I am not sure whether this bug will still exists after the current PR is merged. I prefer to merge the current PR first, and then connect with the issue reporter to see if the bug still exists. We can open a new PR then if needed.

**Review by Jinyu-W — APPROVED** [review]
